src/controllers/activity.controller.js

```diff
@@ -0,0 +1,19 @@
+const Activity = require('../models/Activity.model');
+
+const store = (req, res) => {
+  const {
+    name, hours, startDate, endDate, trainerName, location, ProgramId, ObjectiveId,
+  } = req.body;
+
+  const activity = new Activity({
+    name, hours, startDate, endDate, trainerName, location, ProgramId, ObjectiveId,
+  });
+  activity.save()
+    .then(() => {
+      res.send({ state: true });
```

Comment:
I assume this is the JSON they will be sending back to the front end? and then from the front end you will check the response for the state property and if true, carry on, and if not then, notify the user or try again or something else, sah?

Two things:

- when sending back `JSON` use `res.json` instead of `res.send`

- `state` is very ambiguos for the reciever on the front end. I would call it something like `success` or `saved`. Then if false you can have another one with more info, so:

```js
.then(() => {
  res.json({success:true})
}).catch(err => {
  if (err.message= 'Path `name` is required.'){ // example of a mongo error
    res.json({success:false, message:'Required fields missing'})
  } else {
    res.json({success:false})
  }
}) 

```